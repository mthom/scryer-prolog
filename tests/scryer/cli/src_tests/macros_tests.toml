args = [
    "-f",
    "--no-add-history",
    "-g", "listing(example/1)",
    "-g", "halt",
    "src/tests/macros.pl"
]
