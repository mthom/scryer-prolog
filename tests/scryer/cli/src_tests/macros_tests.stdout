example(formated_string(A)) :-
   A="This is my term foo(1)".
example(bignum_example(A,B,C)) :-
   (  bignum_lt(A,B),
      !,
      bignum_gt(A,0)
   ;  bignum_lt(B+1,C)
   ),
   compute(A,B).
example(numeric_enum_example) :-
   foo(100,101),
   bar([110,111|A]).
